src/main/java/com/treasure_data/client/HttpConnectionImpl.java

```diff
@@ -177,35 +178,42 @@ public void doPostRequest(Request<?> request, String path, Map<String, String> h
         StringBuilder sbuf = new StringBuilder();
         sbuf.append(getSchemeHostPort(System.getenv(Config.TD_ENV_API_SERVER))).append(path);
 
+        URL url = new URL(sbuf.toString());
+        conn = (HttpURLConnection) url.openConnection();
+        conn.setRequestMethod("POST");
+        conn.setReadTimeout(postReadTimeout);
+
+        // header
+        setRequestAuthHeader(request, conn);
+        if (header != null && !header.isEmpty()) {
+            for (Map.Entry<String, String> e : header.entrySet()) {
+                conn.setRequestProperty(e.getKey(), e.getValue());
+            }
+        }
+
         // parameters
         if (params != null && !params.isEmpty()) {
-            sbuf.append("?");
+            StringBuilder queryParam = new StringBuilder();
             int paramSize = params.size();
             Iterator<Map.Entry<String, String>> iter = params.entrySet().iterator();
             for (int i = 0; i < paramSize; ++i) {
                 Map.Entry<String, String> e = iter.next();
-                sbuf.append(e.getKey()).append("=").append(e.getValue());
+                queryParam.append(e.getKey()).append("=").append(e.getValue());
                 if (i + 1 != paramSize) {
-                    sbuf.append("&");
+                    queryParam.append("&");
                 }
             }
-            URL url = new URL(sbuf.toString());
-            conn = (HttpURLConnection) url.openConnection();
+
+            byte[] bodyData = queryParam.toString().getBytes("UTF-8");
```

Comment:
For example, https://github.com/treasure-data/td-client-java/blob/7093e43200a3d9af028e14f2662c5be61b061b10/src/main/java/com/treasure_data/client/DefaultClientAdaptorImpl.java#L1511-L1511
This code performs URL encoding for each value